fit_multiband.py

```diff
@@ -0,0 +1,255 @@
+            A list of catalog-exposure pairs in a given band. Subclasses may require there to only be one catexp per
+            band, and/or for the catalogs to contain HeavyFootprints with deblended images.
+        cat_ref : `lsst.afw.table.SourceCatalog`
+            A reference source catalog to fit. Subclasses may be configured fit only a subset of these sources.
+        kwargs
+            Additional keyword arguments.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with a cat_output attribute containing the output measurement catalog.
+        """
+        raise RuntimeError("Not implemented")
+
+
+class MultibandFitConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=MultibandFitConnections,
+):
+    fit_multiband = pexConfig.ConfigurableField(
+        target=MultibandFitSubTask,
+        doc="Task to fit sources using multiple bands",
+    )
+
+    def get_band_sets(self):
+        dict_fit = self.toDict()['fit_multiband']
+        bands_to_fit = dict_fit.get('bands_fit')
+        if bands_to_fit is None:
+            raise RuntimeError(f'{__class__}.fit_multiband must have bands_fit attribute '
+                               f'(self.toDict()["fit_multiband"]={dict_fit})')
+        return set(bands_to_fit), set(dict_fit.get('bands_read', ()))
+
+
+class MultibandFitTask(pipeBase.PipelineTask):
+    ConfigClass = MultibandFitConfig
+    _DefaultName = "multibandFit"
+
+    def __init__(self, initInputs, **kwargs):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("fit_multiband", schema=initInputs["cat_ref_schema"].schema)
+        self.cat_output_schema = afwTable.SourceCatalog(self.fit_multiband.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        input_objs = [inputRefs.cats_meas, inputRefs.coadds]
+        dataIds_cats, dataIds = [[dRef.dataId for dRef in obj] for obj in input_objs]
+        if not all((x == y for x, y in zip(dataIds, dataIds_cats))):
+            raise RuntimeError(f'Mismatched cats_meas, coadds dataIds {dataIds_cats} {dataIds}')
+        bands_data = [dataId['band'] for dataId in dataIds]
+        bandset_fit, bandset_read = self.config.get_band_sets()
+        bandset_needed = bandset_fit.union(bandset_read)
+        bandset_data = set(bands_data)
+        if bandset_data != bandset_needed:
+            raise RuntimeError(f'set(band={bands_data})={bandset_data} != bandset_needed={bandset_needed}'
+                               f'=(bands_fit={bandset_fit}).union(bands_read={bandset_read}))')
+        cat_ref, cats_meas, coadds = (inputs[x] for x in ('cat_ref', 'cats_meas', 'coadds'))
+        n_data = [len(x) for x in (bands_data, cats_meas, coadds)]
+        if not all(x == n_data[0] for x in n_data[1:]):
+            raise RuntimeError(f'bands, cats_meas, coadds, dataIds lens={n_data} not matched')
```

Comment:
Some comments around all of the logic in this method would be welcome; I could not keep track of what the different lists/sets of bands really meant or how you were relating them.

And if any of these checks are invariants that are guaranteed by `adjustQuantum`, feel free to just assert on them (or even skip them) here.  If they are not guaranteed by `adjustQuantum`, please add some context to the error messages.